src/components/safe-apps/SafeAppIcon/index.tsx

```diff
@@ -1,17 +1,19 @@
 import { type ReactElement, memo } from 'react'
 
-const getIframeContent = (src: string): string => {
-  return `<style>
-    html, body {
-      padding: 0;
-      margin: 0;
-      background-color: transparent;
-      height: 100%;
-    }
-    body {
-      background: url('${src}') center center/contain no-repeat;
-    }
-  </style>`
+const APP_LOGO_FALLBACK_IMAGE = `/images/apps/app-placeholder.svg`
+
+const getIframeContent = (url: string, alt: string, width: number, height: number): string => {
+  return `
+     <body style="margin: 0;">
+       <img src=${url} alt=${alt} width="${width}" height="${height}" />
+       <script>
+          document.querySelector("img").addEventListener("error", (e) => {
+           e.target.onerror = null
```

Comment:
You need to assign the handler like this as well then: `document.querySelector("img").onerror =`

Reply:
I changed it but just fyi it's unclear why I "need to assign" it that way from your message (I assumed stylistic reasons)

Reply:
I meant it won't work like this. Try this in the console, it will log `error` twice.
```
const img = new Image()
img.addEventListener('error', () => {
    console.log('error')
    img.onerror = null
})
img.src = '123'
setTimeout(() => {
    img.src = '456'
}, 100)
```

Reply:
<img width="316" alt="Screenshot 2023-01-11 at 11 08 09" src="https://user-images.githubusercontent.com/16622558/211777883-a5b5a7c1-1c07-4cc0-afc7-932a4c1d9a31.png">

Reply:
Not sure I get it why it wouldn't have worked

Reply:
<img width="290" alt="Screenshot 2023-01-11 at 11 54 53" src="https://user-images.githubusercontent.com/381895/211788614-9fd6192e-da2d-4c38-ad38-325b7ba44502.png">

My understanding is you can't clear an addEventListener handler via an old style DOM1 `onerror = null` assignment.

Reply:
Here it gets a valid URL as a source, so the browser makes a request (hence a failed request to http://0.0.0.1), and after 100ms the src is changed to another URL and the browser makes a second request. I guess the handlers may be cached at the time of the request. If you change the first `img.src` assignment to an invalid URL to guarantee that the handler will be called before the source change, it will log `error` only once.

Reply:
Yes. My point is, you need to either use `addEventListener` + `removeEventListener`, or `onerror` + `onerror = null`, otherwise the handler won't be removed.

Reply:
> My understanding is you can't clear an addEventListener handler via an old style DOM1 onerror = null assignment.

Yeah you seem to be right, if I change the snippet to `onerror`-style assignment it works as expected